src/wallet/wallet2.cpp

```diff
@@ -3737,12 +3737,24 @@ void wallet2::store_to(const std::string &path, const epee::wipeable_string &pas
     }
   } else {
     // save to new file
+#ifdef WIN32
+    // On Windows avoid using std::ofstream which does not work with UTF-8 filenames
+    // The price to pay is temporary higher memory consumption for string stream + binary archive
+    std::ostringstream oss;
+    binary_archive<true> oar(oss);
+    bool success = ::serialization::serialize(oar, cache_file_data);
+    if (success) {
+        success = epee::file_io_utils::save_string_to_file(new_file, oss.str());
+    }
+    THROW_WALLET_EXCEPTION_IF(!success, error::file_save_error, new_file);
+#else
     std::ofstream ostr;
     ostr.open(new_file, std::ios_base::binary | std::ios_base::out | std::ios_base::trunc);
     binary_archive<true> oar(ostr);
     bool success = ::serialization::serialize(oar, cache_file_data);
     ostr.close();
     THROW_WALLET_EXCEPTION_IF(!success || !ostr.good(), error::file_save_error, new_file);
+#endif
```

Comment:
Since the WIN32 code path here isn't actually gunky, I think you can make it the only one, it's simpler and will avoid some new code possibly copy the ofstream code path elsewhere.

Reply:
So in your opinion not to worry about the RAM allocation that gets a temporary spike 50 MB higher or so? I was thinking about people running Monero on Linux on VPS servers with as little RAM as possible to keep costs down. (But then maybe only daemons anyway, not wallets ...)

Reply:
AIUI, the former version using `ostringstream` + `save_string_to_file` and the latter version using `ofstream` would be doing effectively the same thing, even for non-Windows systems. Your concern about the increased RAM usage doesn't seem to apply because the definition of `save_string_to_file` wouldn't change for non-Windows.

Reply:
@stoffu: Ah, now I am confused :)

The way I see it: The solution so far holds around 50 MB of data in `cache_file_data` which the `ofstream`-based serializer will serialize and write out to file piece by piece without needing much more RAM than that.

My new solution for Windows of course also holds 50 MB of data in `cache_file_data` but serializes into the string behind `oss` which will grow to a size of another 50 MB and will be written to file all at once only at the very end.

Maybe I don't know enough how the serializer works in conjunction with a `ofstream`, maybe the data does **not** go to file piece by piece in the "old" solution? And do you mean to say that also the "old" solution uses `save_string_to_file` somehow?

Reply:
Ah, I was wrong then:) There indeed is a difference between the two.

Reply:
I don't think 50 MB is a big problem at all, but if you think it is, I'm OK with it.

Reply:
Thanks, moneromooo, I really feel better letting it in: That way I am sure not to break something; those cache files will grow over time; and we have wallets now on smartphones as well.